observability/opentelemetry/v2/client/otel_observability_service.go

```diff
@@ -0,0 +1,197 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package client
+
+import (
+	"context"
+	"strings"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/observability"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+	cehttp "github.com/cloudevents/sdk-go/v2/protocol/http"
+)
+
+// OTelObservabilityService implements the ObservabilityService interface from cloudevents
+type OTelObservabilityService struct {
+	tracer               trace.Tracer
+	spanAttributesGetter func(cloudevents.Event) []attribute.KeyValue
+	spanNameFormatter    func(cloudevents.Event) string
+}
+
+// NewOTelObservabilityService returns a OpenTelemetry enabled observability service
+func NewOTelObservabilityService(opts ...OTelObservabilityServiceOption) *OTelObservabilityService {
+	tracerProvider := otel.GetTracerProvider()
+
+	os := &OTelObservabilityService{
+		tracer: tracerProvider.Tracer(
+			instrumentationName,
+			trace.WithInstrumentationVersion("1.0.0"), // TODO: Can we have the package version here?
```

Comment:
Since I'm not sure if we can get it at runtime (my search didn't yield any good results) I removed it for now.

Reply:
It would be nice to have though. Maybe they have it stored in a metadata file or constant somewhere. You could leave it commented out but with the TODO so they can tell you.

Reply:
Right. I left the line commented out, hopefully they can point me on how to get it.